server/public/model/config.go

```diff
@@ -4322,49 +4322,110 @@ func (o *Config) GetSanitizeOptions() map[string]bool {
+				}
+			}
+		case string:
+			if shouldSanitize(key, rules) && v != "" {
+				config[key] = FakeSetting
+			}
+		default:
+			continue
+		}
 	}
+	return nil
+}
 
-	if o.FileSettings.PublicLinkSalt != nil {
-		*o.FileSettings.PublicLinkSalt = FakeSetting
+func shouldSanitize(key string, rules SanitizeConfigRules) bool {
+	lowerKey := strings.ToLower(key)
+	for _, substring := range rules.Contains {
+		if strings.Contains(lowerKey, strings.ToLower(substring)) {
+			return true
+		}
+	}
+	for _, prefix := range rules.StartsWith {
+		if strings.HasPrefix(lowerKey, strings.ToLower(prefix)) {
+			return true
+		}
 	}
+	for _, suffix := range rules.EndsWith {
+		if strings.HasSuffix(lowerKey, strings.ToLower(suffix)) {
+			return true
+		}
+	}
+	return false
+}
 
-	if o.FileSettings.AmazonS3SecretAccessKey != nil && *o.FileSettings.AmazonS3SecretAccessKey != "" {
-		*o.FileSettings.AmazonS3SecretAccessKey = FakeSetting
+func (o *Config) Sanitize() *AppError {
+	rules := SanitizeConfigRules{
+		Contains:   []string{"password", "secret", "key", "keyid", "email"},
+		StartsWith: []string{"key", "password", "email"},
+		EndsWith:   []string{"key", "keyid", "id", "email"},
 	}
 
-	if o.EmailSettings.SMTPPassword != nil && *o.EmailSettings.SMTPPassword != "" {
-		*o.EmailSettings.SMTPPassword = FakeSetting
+	// Marshal the config struct to JSON and back to map
+	var configMap map[string]interface{}
+	data, err := json.Marshal(o)
+	if err != nil {
+		return NewAppError("Config.Sanitize", "model.config.sanitize.marshal.app_error", nil, "", http.StatusInternalServerError)
 	}
 
-	if o.GitLabSettings.Secret != nil && *o.GitLabSettings.Secret != "" {
-		*o.GitLabSettings.Secret = FakeSetting
+	if err := json.Unmarshal(data, &configMap); err != nil {
+		return NewAppError("Config.Sanitize", "model.config.sanitize.unmarshal_to_map.app_error", nil, "", http.StatusInternalServerError)
 	}
 
-	if o.GoogleSettings.Secret != nil && *o.GoogleSettings.Secret != "" {
-		*o.GoogleSettings.Secret = FakeSetting
+	// Sanitize the map
+	if err := sanitizeConfigMap(configMap, rules); err != nil {
+		return err
 	}
 
-	if o.Office365Settings.Secret != nil && *o.Office365Settings.Secret != "" {
-		*o.Office365Settings.Secret = FakeSetting
+	// Marshal the map back to JSON and into struct
+	sanitizedData, err := json.Marshal(configMap)
+	if err != nil {
+		return NewAppError("Config.Sanitize", "model.config.sanitize.marshal.app_error", nil, "", http.StatusInternalServerError)
 	}
 
-	if o.OpenIdSettings.Secret != nil && *o.OpenIdSettings.Secret != "" {
-		*o.OpenIdSettings.Secret = FakeSetting
+	if err := json.Unmarshal(sanitizedData, o); err != nil {
+		return NewAppError("Config.Sanitize", "model.config.sanitize.unmarshal_to_map.app_error", nil, "", http.StatusInternalServerError)
 	}
 
-	if o.SqlSettings.DataSource != nil {
-		*o.SqlSettings.DataSource = FakeSetting
+	// Sanitize other fields not matching the sanitization rules
+	if o.FileSettings.PublicLinkSalt != nil {
```

Comment:
there will continue being exceptions to some configs that the name does not strictively matches the sensitive named rules but we still want to obfuscate, those will continue being added in this same way.